### PR TITLE
Replace Custom Generator Prompt With Inquirer

### PR DESCRIPTION
**Pull request by NickGerleman**
We have a chunk of code to syncrhonously read "y" or "n" when prompting users to ask if we should replace their existing files. @react-native-community/cli already includes inquirer for prompting duties. Add inquirer to our deps as well and use it instead of rolling our own logic.

Note that this means moving a lot of this code to be async. This is mostly mechanical, but changes the generator boundary, meaning we need to add more fallback logic to react-native-windows init.

Haven't validated this yet, but will test against master and 0.62 before merging.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5417)

### CODE REVIEW

**Comment by NickGerleman** [comment]
Confirmed `react-native-windows-init` works with both canary and 0.62.

**Comment by dannyvv** [comment]
> ) {

```suggestion
) : Promise<void> {
```

---
Refers to: vnext/local-cli/src/generator-common/index.ts:86 in c0fa3c3. [](commit_id = c0fa3c3652360042a803daa6f71aa4e2e99cb64e, deletion_comment = False)

**Comment by dannyvv** [comment]
>   const contentChangedCallback: ContentChangedCallback = alwaysOverwrite

future improvement: Should we just inline `upgradeFileContentChangedCallback` in this method and deal with alwaysOverwrite before the prompt?

---
Refers to: vnext/local-cli/src/generator-common/index.ts:205 in c0fa3c3. [](commit_id = c0fa3c3652360042a803daa6f71aa4e2e99cb64e, deletion_comment = False)

File: vnext/local-cli/src/generator-common/index.ts [hunk 1]
```diff
@@ -351,39 +216,39 @@ export function copyAndReplaceWithChangedCallback(
           contentChanged,
         );
```
Comment (dannyvv) [inline vnext/local-cli/src/generator-common/index.ts:217]: Future cleanup: Now that you have made the prompt case is a lot cleaner, should we merge these tow methods and pass in `alwaysOverwrite`?

**Comment by NickGerleman** [comment]
> > ) {
> 
> Not sure about the style-guides if `async` is enough or if we want to be explicit on the return type of:
> 
> ```
> ) : Promise<void> {
> ```
> 
> Note, also applicable to the other locations in this file.
> 
> Refers to: vnext/local-cli/src/generator-common/index.ts:86 in [c0fa3c3](https://github.com/microsoft/react-native-windows/commit/c0fa3c3652360042a803daa6f71aa4e2e99cb64e). [](commit_id = [c0fa3c3](https://github.com/microsoft/react-native-windows/commit/c0fa3c3652360042a803daa6f71aa4e2e99cb64e), deletion_comment = False)

I feel like I've seen lack of explicit return type more often than Promise<void>, but yeah, there hasn't been much of an enforced style guide. Not sure if TS has anything official here.

**Pull request merged**